besskge/scoring.py

```diff
@@ -1175,3 +1175,150 @@ def score_tails(
             width_ht.view(-1, 1, 2, self.embedding_size),
             box_size.view(-1, 1, 2),
         )
+
+
+class InterHT(DistanceBasedScoreFunction):
+    """
+    InterHT scoring function :cite:p:`InterHT`.
+    """
+
+    def __init__(
+        self,
+        negative_sample_sharing: bool,
+        scoring_norm: int,
+        sharding: Sharding,
+        n_relation_type: int,
+        embedding_size: int,
+        entity_initializer: Union[torch.Tensor, List[Callable[..., torch.Tensor]]] = [
+            init_KGE_uniform
+        ],
+        relation_initializer: Union[torch.Tensor, List[Callable[..., torch.Tensor]]] = [
+            init_KGE_uniform
+        ],
+        normalize_entities: bool = True,
+        offset: float = 1.0,
+    ) -> None:
+        """
+        Initialize InterHT model.
+
+        :param negative_sample_sharing:
+            see :meth:`DistanceBasedScoreFunction.__init__`
+        :param scoring_norm:
+            see :meth:`DistanceBasedScoreFunction.__init__`
+        :param sharding:
+            Entity sharding.
+        :param n_relation_type:
+            Number of relation types in the knowledge graph.
+        :param embedding_size:
+            Size of entity and relation embeddings.
+        :param entity_initializer:
+            Initialization function or table for entity embeddings.
+        :param relation_initializer:
+            Initialization function or table for relation embeddings.
+        :param normalize_entities:
+            If True, L2-normalize head and tail entity embeddings before projecting.
```

Comment:
Multiplying by the auxiliary embeddings mathematically is not dissimilar to PairRE, so it's indeed a projection, but in the InterHT paper they don't use this expression explicitly I think - maybe worth reformulating this for clarity? Not sure though.